easybuild/tools/toolchain/compiler.py

```diff
@@ -239,23 +239,34 @@ def _set_compiler_vars(self):
 
     def _set_compiler_flags(self):
         """Collect the flags set, and add them as variables too"""
-
-        flags = [self.options.option(x) for x in self.COMPILER_FLAGS if self.options.get(x, False)]
-        cflags = [self.options.option(x) for x in self.COMPILER_C_FLAGS + self.COMPILER_C_UNIQUE_FLAGS
+        variants = ['', '_F', '_F_UNIQUE', '_C', '_C_UNIQUE', '_OPT', '_PREC']
+        for variant in variants:
+            old_var = getattr(self, f'COMPILER{variant}_FLAGS', None)
+            if old_var is not None:
+                self.log.deprecated(f'COMPILER{variant}_FLAGS has been renamed to COMPILER{variant}_OPTIONS.', '6.0')
+                setattr(self, f'COMPILER{variant}_OPTIONS', old_var)
```

Comment:
Added warnings and backwards compat to handle the renamed variables to address @boegel review